File: SDET_Question_10/SDET_Script/Makemytrip.py
# from telnetlib import EC

import pytest
from selenium import webdriver
from selenium.webdriver import Keys
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.support import expected_conditions as EC


@pytest.fixture
def driver():
    # Set up Chrome WebDriver
    driver = webdriver.Chrome(service=Service(ChromeDriverManager().install()))
    yield driver
    driver.quit()


def test_makemytrip_flight_search(driver):
    # Open the MakeMyTrip website
    driver.get("https://www.makemytrip.com/")

    # Click on "Flights" tab
    wait= WebDriverWait(driver, 25)
    pop_up=wait.until(EC.element_to_be_clickable((By.XPATH, "//span[@data-cy='closeModal']")))
    # driver.find_element(By.XPATH("//span[@data-cy='closeModal']"))
    pop_up.click()
    flights_tab = wait.until(EC.element_to_be_clickable((By.XPATH, "//li[@data-cy='menu_Flights']")))
    flights_tab.click()

    # Select "Round Trip"
    round_trip_button = wait.until(EC.element_to_be_clickable((By.XPATH, "//li[@data-cy='roundTrip']")))
    round_trip_button.click()

    # Enter "FROM" location as "HYD"
    from_location_input = wait.until(EC.presence_of_element_located((By.ID, "fromCity")))
    # from_location_input.clear()
    from_location_input.send_keys("HYD")
    from_location_input.send_keys(Keys.RETURN)

    # Enter "TO" location as "MAA"
    to_location_input = wait.until(EC.presence_of_element_located((By.ID, "toCity")))
    # to_location_input.clear()
    to_location_input.send_keys("MAA")
    to_select_city = wait.until(EC.element_to_be_clickable((By.XPATH, "//p[text()='Chennai, India']")))
    to_select_city.click();

    # Select Departure Date (Assuming the date picker is visible)
    departure_date_xpath="//div[@aria-label='Sun Nov 19 2023']"
    departure_date_picker = wait.until(EC.element_to_be_clickable((By.XPATH, departure_date_xpath)))
    departure_date_picker.click()

    # TODO: Choose a departure date from the date picker

    # Select Return Date (Assuming the date picker is visible)
    return_date_XPATH="//div[@aria-label='Fri Dec 01 2023']"
    return_date_picker = wait.until(EC.element_to_be_clickable((By.XPATH, return_date_XPATH)))
    return_date_picker.click()

    # TODO: Choose a return date from the date picker

    # Click on the "Search" button
    search_button = wait.until(EC.presence_of_element_located((By.XPATH, "//a[text()='Search']")))
    search_button.click()

    # Wait for the search page to load (you can customize the timeout)
    wait.until(EC.title_contains("MakeMyTrip"))

    # Verify that the search page is displayed as expected
    assert "MakeMyTrip" in driver.title

    if __name__=="__main__":
        pytest.main()

File: SDET_Question_3/src/test/java/MakeMyTrip.java
import static org.testng.Assert.assertTrue;

import java.time.Duration;
import java.util.concurrent.TimeUnit;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.annotations.AfterMethod;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import io.github.bonigarcia.wdm.WebDriverManager;

public class MakeMyTrip {

	WebDriver driver;
	WebDriverWait wait;
	 
	@BeforeMethod
	public void LaunchBrowser() {
		WebDriverManager.chromedriver().setup();
		   driver=new ChromeDriver();
		   driver.manage().window().maximize();
		   driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
		   wait= new WebDriverWait(driver,Duration.ofSeconds(25));
		   
	}
	
	@Test
	public void SearchFlight() {
		try {
			
		driver.get("https://www.makemytrip.com/");

        WebElement popUP = wait.until(ExpectedConditions.presenceOfElementLocated(By.xpath("//span[@data-cy='closeModal']")));
        popUP.click();
        
        WebElement flightsTab = driver.findElement(By.xpath("//li[@data-cy='menu_Flights']"));
        flightsTab.click();

        // Select ROUND TRIP
        WebElement roundTripButton = driver.findElement(By.xpath("//li[@data-cy='roundTrip']"));
        roundTripButton.click();

        // Enter From Location as HYD
        WebElement fromLocation = driver.findElement(By.id("fromCity"));
        fromLocation.sendKeys("HYD");

        // Select the first option from the auto-suggest dropdown
        WebElement autoSuggestOption = driver.findElement(By.xpath("//p[contains(text(),'Hyderabad, India')]"));
        autoSuggestOption.click();

        // Enter To Location as MAA
        WebElement toLocation = driver.findElement(By.id("toCity"));
        toLocation.sendKeys("MAA");

        // Select the first option from the auto-suggest dropdown
        autoSuggestOption = driver.findElement(By.xpath("//p[contains(text(),'Chennai, India')]"));
        autoSuggestOption.click();

        // Select DEPARTURE Date (Assuming the input field has an ID)
        String departure_date_xpath="//div[@aria-label='Sun Nov 19 2023']";
        WebElement departureDate = driver.findElement(By.xpath(departure_date_xpath));
        departureDate.click();
        
        // Select RETURN Date (Assuming the input field has an ID)
        String return_date_XPATH="//div[@aria-label='Fri Dec 01 2023']";
        WebElement returnDate = driver.findElement(By.xpath(return_date_XPATH));
        returnDate.click();

        // Click on Search Button
        WebElement searchButton =wait.until(ExpectedConditions.presenceOfElementLocated(By.xpath("//a[text()='Search']")));
        searchButton.click();

        WebElement okay_button=wait.until(ExpectedConditions.presenceOfElementLocated(By.xpath("//button[text()='OKAY, GOT IT!']")));
        okay_button.click();
        // Verify the Search page is displayed as expected (You can add more verification as needed)
        WebElement searchResult = driver.findElement(By.xpath("//span[text()='SEARCH']"));
        String title=driver.getTitle();
        if(title.equalsIgnoreCase("MakeMyTrip")) {
        if (searchResult.isDisplayed()) {
            System.out.println("Search page displayed as expected.");
            assertTrue(true);
        } else {
            System.out.println("Search page not displayed as expected.");
            assertTrue(false);
        }
        }

    }catch (Exception e) {
        System.out.println("Exception caught");
    }

	}
	
	@AfterMethod
	public void TearDown() {
		driver.quit();
	}
}
